### ISSUE TITLE
Fatal Python error: init_fs_encoding on Terminal startup

### ISSUE THREAD

**Issue by kristanner2015** [VSCodium/vscodium#760]
**Describe the bug**
On first launch of the application from the appimage (./VSCodium.AppImage) when the terminal window opens an error mesage is displayed.  The error displayed is as follows:
```
Python path configuration:
  PYTHONHOME = '/tmp/.mount_VSCodiyCMxJV/usr/'
  PYTHONPATH = '/tmp/.mount_VSCodiyCMxJV/usr/share/pyshared/:/usr/bin/python3.9'
  program name = '/usr/bin/python3.9'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/usr/bin/python3.9'
  sys.base_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.base_exec_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.platlibdir = 'lib'
  sys.executable = '/usr/bin/python3.9'
  sys.prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.exec_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.path = [
    '/tmp/.mount_VSCodiyCMxJV/usr/share/pyshared/',
    '/usr/bin/python3.9',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python39.zip',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python3.9',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python3.9/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007f85d360f740 (most recent call first):
<no Python frame>
Python path configuration:
  PYTHONHOME = '/tmp/.mount_VSCodiyCMxJV/usr/'
  PYTHONPATH = '/tmp/.mount_VSCodiyCMxJV/usr/share/pyshared/:/usr/bin/python3.9'
  program name = '/usr/bin/python3.9'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/usr/bin/python3.9'
  sys.base_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.base_exec_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.platlibdir = 'lib'
  sys.executable = '/usr/bin/python3.9'
  sys.prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.exec_prefix = '/tmp/.mount_VSCodiyCMxJV/usr'
  sys.path = [
    '/tmp/.mount_VSCodiyCMxJV/usr/share/pyshared/',
    '/usr/bin/python3.9',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python39.zip',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python3.9',
    '/tmp/.mount_VSCodiyCMxJV/usr/lib/python3.9/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007fb2c4cf8740 (most recent call first):
<no Python frame>
```

**Please confirm that this problem is VSCodium-specific**
- [ ] This bug doesn't happen if I use Microsoft's Visual Studio Code. It only happens in VSCodium.
I cannot test the VSCode behaviour as there is no AppImage for VS Code.

**Please confirm that the issue/resolution isn't already documented**
- [x] I checked [the Docs page](https://github.com/VSCodium/vscodium/blob/master/DOCS.md) and my issue is not mentioned there.

**To Reproduce**
1. Launch AppImage
2. View in-app Terminal window
3. See error

**Expected behavior**
There should not be an error finding python.  My system has a fully functioning python3.9 setup and I use python for many things outside of VSCodium

**Screenshots**

**Desktop (please complete the following information):**
 - OS: gentoo Linux
 - Architecture: x64
 - Version: VSCodium-1.58.0-1625926890.glibc2.17-x86_64.AppImage

**Additional context**
![2021-07-11-165915_2637x1621_scrot](https://user-images.githubusercontent.com/10699763/125213987-9c7fd980-e269-11eb-80cf-0c93d3cc10da.png)

**Comment by daiyam** [comment]
I've also the issue on my Arch Linux with Python managed with asdf. Not sure how to fix that...

**Comment by A-G-D** [comment]
Same problem here. This happened to me when I updated vscodium AppImage from 1.56.X to the latest build (1.58.2) few days ago. Python is otherwise fully functional outside the vscodium terminal.

**Comment by daiyam** [comment]
@A-G-D Yep. `1.57` was updated with the latest AppImage builder to fix #330...

**Comment by A-G-D** [comment]
@daiyam I don't think that's the same issue here. The vscodium itself doesn't crash (unlike in your linked issue) - only that the terminal fails to setup(?) environment for python.

**Comment by daiyam** [comment]
@A-G-D I know but this new issue must be related to the change (that I've made)...

**Comment by daiyam** [comment]
@A-G-D  @kristanner2015 Can you test following AppImage [VSCodium-1.58.2-1626977762.glibc2.33-x86_64.AppImage.zip](https://app.box.com/s/xucb2njq3fst9f5hdccehjpljrnlnhg7)?

**Comment by kristanner2015** [comment]
@daiyam  - I tried that version but it gives a totally different error.  Launched from a terminal window I see a bunch of errors like:
`Error: /usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_VSCodiQnduqk/usr/share/codium/resources/app/node_modules.asar.unpacked/spdlog/build/Release/spdlog.node)`

Then when the window opens all input is blocked by a modal dialog asking about trusting the project folder I had open last.  It does not accept any input from keyboard or mouse.
I then hit Ctrl-C in the terminal window  and that causes another message to pop up in the GUI about the extension host terminating unxpectedly.  See attached screenshot and console output.

I have disabled all extensions except the built-in python extension and a Jupiter notebook extension and the same errors still occur.
[VScodium_error_log-2021.07.22.txt](https://github.com/VSCodium/vscodium/files/6865559/VScodium_error_log-2021.07.22.txt)
![2021-07-22-153103_2650x1687_scrot](https://user-images.githubusercontent.com/10699763/126717920-3a8a6e4b-6081-444b-8733-904f69c9af7d.png)

**Comment by kristanner2015** [comment]
Just checked - my gentoo installation only has glibcxx_3.4.28 at the moment.  No updates available at the moment, but I'll try again as soon as I see one.

**Comment by daiyam** [comment]
@kristanner2015 Sorry about that, I've built it with my Arch so it's using the latest `glibc`. I'm rebuilding the image with the build-agent in docker to have the same version as the releases.

**Comment by daiyam** [comment]
Here the good version: [VSCodium-1.58.2-1626996970.glibc2.17-x86_64.AppImage](https://app.box.com/s/xbuk1wey5ok335bsn5sp05ocs2z6k76i)

**Comment by A-G-D** [comment]
@daiyam Using that version, the error seem to be fixed now.

**Comment by kristanner2015** [comment]
I can confirm.  With the 1.58.2-162996950.glibc2.17 version python is working again.
